crates/router/src/routes/payment_methods.rs

```diff
@@ -224,6 +229,11 @@ pub async fn payment_method_retrieve_api(
     })
     .into_inner();
 
+    let auth = match auth::is_ephemeral_auth(req.headers()) {
```

Comment:
What is reason to make retrieve as ephemeral_auth?

Remove ephemeral_auth, this should be only api-key based auth.